x/finality/keeper/power_dist_change.go

```diff
@@ -91,47 +91,80 @@ func (k Keeper) RecordVotingPowerAndCache(ctx context.Context, newDc *ftypes.Vot
 }
 
 // HandleFPStateUpdates emits events and triggers hooks for finality providers with state updates
-func (k Keeper) HandleFPStateUpdates(ctx context.Context, prevDc, newDc *ftypes.VotingPowerDistCache, state *ftypes.ProcessingState) {
+func (k Keeper) HandleFPStateUpdates(
+	ctx context.Context,
+	state *ftypes.ProcessingState,
+	prevDc, newDc *ftypes.VotingPowerDistCache,
+	shouldCallHooks bool,
+) {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
 
 	newlyActiveFPs := newDc.FindNewActiveFinalityProviders(prevDc)
 	for _, fp := range newlyActiveFPs {
-		if err := k.HandleActivatedFinalityProvider(ctx, fp.BtcPk); err != nil {
-			panic(fmt.Errorf("failed to execute after finality provider %s activated", fp.BtcPk.MarshalHex()))
-		}
+		k.processActiveFp(sdkCtx, fp, shouldCallHooks)
+	}
 
-		statusChangeEvent := types.NewFinalityProviderStatusChangeEvent(fp.BtcPk, types.FinalityProviderStatus_FINALITY_PROVIDER_STATUS_ACTIVE)
-		if err := sdkCtx.EventManager().EmitTypedEvent(statusChangeEvent); err != nil {
-			panic(fmt.Errorf(
-				"failed to emit FinalityProviderStatusChangeEvent with status %s: %w",
-				types.FinalityProviderStatus_FINALITY_PROVIDER_STATUS_ACTIVE.String(), err))
-		}
+	newlyInactiveFPs := newDc.FindNewInactiveFinalityProviders(prevDc)
```

Comment:
Could you double check that `newDc.FindNewInactiveFinalityProviders(prevDc)` returns all new inactive finality providers ?

Asking as I remember the function inside, for some reason, do not reutrn jailed or slashed ones i.e:
```go
func (dc *VotingPowerDistCache) GetInactiveFinalityProviderSet() map[string]*FinalityProviderDistInfo {
	numActiveFPs := dc.NumActiveFps

	if len(dc.FinalityProviders) <= int(numActiveFPs) {
		return nil
	}

	inactiveFps := make(map[string]*FinalityProviderDistInfo)

	for _, fp := range dc.FinalityProviders[numActiveFPs:] {
		if !fp.IsSlashed && !fp.IsJailed {
			inactiveFps[fp.BtcPk.MarshalHex()] = fp
		}
	}

	return inactiveFps
}
```
and our hook:
```go
err := k.hooks.AfterBbnFpRemovedFromActiveSet(ctx, fpAddr)
```
must be called on every new inactive FP.